src/h2olog/h2olog.cc

```diff
@@ -189,18 +199,28 @@ int main(int argc, char **argv)
         tracer.reset(create_http_tracer());
     }
 
+    std::unordered_set<std::string> available_usdt_names;
+    for (const auto &usdt : tracer->usdt_probes()) {
+        available_usdt_names.insert(usdt.provider + ":" + usdt.name);
+    }
+
     int debug = 0;
+    bool list_and_exit = false;
     FILE *outfp = stdout;
     std::vector<std::string> event_type_filters;
     std::vector<std::string> response_header_filters;
     int c;
     pid_t h2o_pid = -1;
-    while ((c = getopt(argc, argv, "hdp:t:s:w:")) != -1) {
+    while ((c = getopt(argc, argv, "hdp:t:s:w:l")) != -1) {
         switch (c) {
         case 'p':
             h2o_pid = atoi(optarg);
             break;
         case 't':
+            if (available_usdt_names.find(optarg) == available_usdt_names.cend()) {
+                fprintf(stderr, "No such event type: %s\n", optarg);
+                exit(EXIT_FAILURE);
+            }
```

Comment:
... and when there is an matching entry in `available_probes`, then transplant that entry of `available_probes` to `probes`.